test/scaling/scaling.jl

```diff
@@ -11,12 +11,25 @@ using Test, LinearAlgebra, StaticArrays
     @test typeof(sitp) <: Interpolations.ScaledInterpolation
     @test parent(sitp) === itp
 
-    for (x,y) in zip(-3:.05:1.5, 1:.1:10, LinRange(-3,1.5,91))
+    for (x,y) in zip(-3:.05:1.5, 1:.1:10,)
         @test sitp(x) ≈ y
     end
 
     @test_throws ArgumentError scale(itp, reverse(-3:.5:1.5))
 
+    # Model linear interpolation of y = -3 + .5x by interpolating y=x
+    # and then scaling to the new x range, this time using a LinRange,
+    # which should give the same values as with the StepRangeLen used previously
+
+    itp = interpolate(1:1.0:10, BSpline(Linear()))
+
+    sitp = @inferred(scale(itp, LinRange(-3,1.5,91)))
```

Comment:
This duplicates most of the previous test but with a different LinRange.
possibly this is wasteful? But it does fail without the fix as desired.